STUDIO/org.openl.rules.webstudio/src/org/openl/rules/webstudio/mail/MailSender.java

```diff
@@ -0,0 +1,84 @@
+package org.openl.rules.webstudio.mail;
+
+import org.openl.util.IOUtils;
+import org.openl.util.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.URLName;
+import javax.mail.internet.MimeMessage;
+import java.io.IOException;
+import java.util.Properties;
+
+public class MailSender {
+    private static final Logger LOGGER = LoggerFactory.getLogger(MailSender.class);
+
+    private static final String MAIL_VERIFICATION_SUBJECT = "WebStudio mail verification";
+    private static final String MAIL_VERIFICATION_TEMPLATE = "/mail/template/mail-verification.html";
+    private static final String MAIL_VERIFICATION_LINK_PLACEHOLDER = "${link}";
+    private static final String MAIL_VERIFICATION_PLAIN_TEXT = "Please verify you email to proceed with WebStudio: ";
+
+    @Value("${mail.url}")
+    private String url;
+
+    @Value("{mail.username}")
+    private String user;
+
+    @Value("${mail.password}")
+    private String password;
+
+    /**
+     * Send e-mail for verification
+     *
+     * @param to               - recipient
+     * @param verificationLink - Links in the message body
+     * @return true - if e-mail has been sent successful, false - if the service is off.
+     */
+    public boolean sendVerification(String to, String verificationLink) {
+        try {
+            if (StringUtils.isBlank(url) || StringUtils.isBlank(user) || StringUtils.isBlank(password)) {
+                return false;

```

Comment:
It can be replaced with conditional beans:

1. NoMailSender bean which does noting
2. RealMailSender bean